x-pack/plugins/cases/public/api/decoders.ts

```diff
@@ -41,10 +41,8 @@ export const decodeCasesMetricsResponse = (metrics?: CasesMetricsResponse) =>
     fold(throwErrors(createToasterPlainError), identity)
   );
 
-export const decodeCasesBulkGetResponse = <Field extends keyof Case = keyof Case>(
-  res: CasesBulkGetResponseCertainFields<Field>,
-  fields?: string[]
-) => {
+export const decodeCasesBulkGetResponse = (res: CasesBulkGetResponse) => {
+  const fields = Object.keys(CasesBulkGetResponseFieldsRt.props);
   const typeToDecode = getTypeForCertainFieldsFromArray(CasesRt, fields);
```

Comment:
This feels a bit overly complicated. If we always return the same fields and we already have their types we don't need to generate `fields` and then create `typeToDecode` we might as well store that in a const and do away with the extra logic.

Reply:
I think something like `rt.array(CasesBulkGetResponseFieldsRt).decode(res.cases)` might work here.

Reply:
Good point!